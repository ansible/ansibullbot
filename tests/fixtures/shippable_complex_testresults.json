[{"permissions":33188,"path":"/testresults/junit.xml","contents":"<?xml version=\"1.0\" encoding=\"utf-8\"?>\n<testsuites errors=\"6\" failures=\"5\" tests=\"18\" time=\"15.6\">\n\t<testsuite errors=\"1\" failures=\"0\" name=\"another-suite\" skipped=\"0\" tests=\"1\" time=\"0\">\n\t\t<testcase classname=\"another-class\" name=\"another-name\">\n\t\t\t<error message=\"another error message\" type=\"error\">\nanother\nerror\noutput</error>\n\t\t</testcase>\n\t</testsuite>\n\t<testsuite errors=\"5\" failures=\"5\" hostname=\"hostname\" id=\"id\" name=\"test-suite-name\" package=\"package\" skipped=\"5\" tests=\"17\" time=\"15.6\" timestamp=\"2017-03-03T20:30:50\">\n\t\t<properties>\n\t\t\t<property name=\"yes\" value=\"no\"/>\n\t\t\t<property name=\"hello\" value=\"world\"/>\n\t\t</properties>\n\t\t<testcase classname=\"classname\" name=\"test-case-name\" time=\"1.200000\">\n\t\t\t<system-out>stdout</system-out>\n\t\t\t<system-err>stderr</system-err>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"test-case-name-error\" time=\"1.200000\">\n\t\t\t<error message=\"error\nmessage\" type=\"error\">\nerror\noutput</error>\n\t\t\t<system-out>stdout</system-out>\n\t\t\t<system-err>stderr</system-err>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"test-case-name-failure\" time=\"1.200000\">\n\t\t\t<failure message=\"failure\nmessage\" type=\"failure\">\nfailure\noutput</failure>\n\t\t\t<system-out>stdout</system-out>\n\t\t\t<system-err>stderr</system-err>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"test-case-name-skipped\" time=\"1.200000\">\n\t\t\t<skipped message=\"skipped\nmessage\" type=\"skipped\">\nskipped\noutput</skipped>\n\t\t\t<system-out>stdout</system-out>\n\t\t\t<system-err>stderr</system-err>\n\t\t</testcase>\n\t\t<testcase classname=\"class\nname\" name=\"test-case-name-error-no-\noutput\" time=\"1.200000\">\n\t\t\t<error message=\"error message\" type=\"error\"/>\n\t\t\t<system-out>std\nout</system-out>\n\t\t\t<system-err>\nstd\nerr</system-err>\n\t\t</testcase>\n\t\t<testcase classname=\"class\nname\" name=\"test-case-name-failure-no-\noutput\" time=\"1.200000\">\n\t\t\t<failure message=\"failure message\" type=\"failure\"/>\n\t\t\t<system-out>std\nout</system-out>\n\t\t\t<system-err>\nstd\nerr</system-err>\n\t\t</testcase>\n\t\t<testcase classname=\"class\nname\" name=\"test-case-name-skipped-no-\noutput\" time=\"1.200000\">\n\t\t\t<skipped message=\"skipped message\" type=\"skipped\"/>\n\t\t\t<system-out>std\nout</system-out>\n\t\t\t<system-err>\nstd\nerr</system-err>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"test-case-name-error-no-message\" time=\"1.200000\">\n\t\t\t<error type=\"error\">error output</error>\n\t\t\t<system-out>stdout</system-out>\n\t\t\t<system-err>stderr</system-err>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"test-case-name-failure-no-message\" time=\"1.200000\">\n\t\t\t<failure type=\"failure\">failure output</failure>\n\t\t\t<system-out>stdout</system-out>\n\t\t\t<system-err>stderr</system-err>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"test-case-name-skipped-no-message\" time=\"1.200000\">\n\t\t\t<skipped type=\"skipped\">skipped output</skipped>\n\t\t\t<system-out>stdout</system-out>\n\t\t\t<system-err>stderr</system-err>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"test-case-name-error-no-stderr\" time=\"1.200000\">\n\t\t\t<error message=\"error message\" type=\"error\"/>\n\t\t\t<system-out>stdout</system-out>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"test-case-name-failure-no-stderr\" time=\"1.200000\">\n\t\t\t<failure message=\"failure message\" type=\"failure\"/>\n\t\t\t<system-out>stdout</system-out>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"test-case-name-skipped-no-stderr\" time=\"1.200000\">\n\t\t\t<skipped message=\"skipped message\" type=\"skipped\"/>\n\t\t\t<system-out>stdout</system-out>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"optimal-success\"/>\n\t\t<testcase classname=\"classname\" name=\"optimal-error\">\n\t\t\t<error message=\"optimal error message\" type=\"error\">\noptimal\nerror\noutput</error>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"optimal-failure\">\n\t\t\t<failure message=\"optimal failure message\" type=\"failure\">\noptimal\nfailure\noutput</failure>\n\t\t</testcase>\n\t\t<testcase classname=\"classname\" name=\"optimal-skipped\">\n\t\t\t<skipped message=\"optimal skipped message\" type=\"skipped\">\noptimal\nskipped\noutput</skipped>\n\t\t</testcase>\n\t</testsuite>\n</testsuites>\n"},{"permissions":33188,"path":"/testresults/not-a-test.json","contents":"{\"hello\":\"world\",\"number\":9}\n"},{"permissions":33188,"path":"/testresults.json","contents":"{\"total\":18,\"totalPassing\":2,\"totalFailures\":5,\"totalErrors\":6,\"totalSkipped\":5,\"durationSeconds\":15.599999999999996,\"errorDetails\":[{\"testName\":\"another-name\",\"className\":\"another-class\",\"suiteName\":\"another-suite\",\"message\":\"another error message\",\"kind\":\"error\",\"full\":\"\\nanother\\nerror\\noutput\"},{\"testName\":\"test-case-name-error\",\"className\":\"classname\",\"suiteName\":\"test-suite-name\",\"message\":\"error\\nmessage\",\"kind\":\"error\",\"full\":\"\\nerror\\noutput\"},{\"testName\":\"test-case-name-error-no-\\noutput\",\"className\":\"class\\nname\",\"suiteName\":\"test-suite-name\",\"message\":\"error message\",\"kind\":\"error\",\"full\":\"\\nstd\\nerr\"},{\"testName\":\"test-case-name-error-no-message\",\"className\":\"classname\",\"suiteName\":\"test-suite-name\",\"message\":\"\",\"kind\":\"error\",\"full\":\"error output\"},{\"testName\":\"test-case-name-error-no-stderr\",\"className\":\"classname\",\"suiteName\":\"test-suite-name\",\"message\":\"error message\",\"kind\":\"error\",\"full\":\"\"},{\"testName\":\"optimal-error\",\"className\":\"classname\",\"suiteName\":\"test-suite-name\",\"message\":\"optimal error message\",\"kind\":\"error\",\"full\":\"\\noptimal\\nerror\\noutput\"}],\"failureDetails\":[{\"testName\":\"test-case-name-failure\",\"className\":\"classname\",\"suiteName\":\"test-suite-name\",\"message\":\"failure\\nmessage\",\"kind\":\"failure\",\"full\":\"\\nfailure\\noutput\"},{\"testName\":\"test-case-name-failure-no-\\noutput\",\"className\":\"class\\nname\",\"suiteName\":\"test-suite-name\",\"message\":\"failure message\",\"kind\":\"failure\",\"full\":\"\\nstd\\nerr\"},{\"testName\":\"test-case-name-failure-no-message\",\"className\":\"classname\",\"suiteName\":\"test-suite-name\",\"message\":\"\",\"kind\":\"failure\",\"full\":\"failure output\"},{\"testName\":\"test-case-name-failure-no-stderr\",\"className\":\"classname\",\"suiteName\":\"test-suite-name\",\"message\":\"failure message\",\"kind\":\"failure\",\"full\":\"\"},{\"testName\":\"optimal-failure\",\"className\":\"classname\",\"suiteName\":\"test-suite-name\",\"message\":\"optimal failure message\",\"kind\":\"failure\",\"full\":\"\\noptimal\\nfailure\\noutput\"}]}"}]
