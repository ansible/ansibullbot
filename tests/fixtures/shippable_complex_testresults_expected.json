{
  "errors": "6", 
  "failures": "5", 
  "tests": "18", 
  "testsuites": [
    {
      "errors": "1", 
      "failures": "0", 
      "name": "another-suite", 
      "skipped": "0", 
      "testcases": [
        {
          "classname": "another-class", 
          "error": {
            "message": "another error message", 
            "type": "error"
          }, 
          "jobid": null, 
          "name": "another-name"
        }
      ], 
      "tests": "1", 
      "time": "0"
    }, 
    {
      "errors": "5", 
      "failures": "5", 
      "hostname": "hostname", 
      "id": "id", 
      "name": "test-suite-name", 
      "package": "package", 
      "properties": {
        "hello": "world", 
        "yes": "no"
      }, 
      "skipped": "5", 
      "testcases": [
        {
          "classname": "classname", 
          "jobid": null, 
          "name": "test-case-name", 
          "system-out": {
            "text": "stdout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "classname", 
          "error": {
            "message": "error message", 
            "type": "error"
          }, 
          "jobid": null, 
          "name": "test-case-name-error", 
          "system-out": {
            "text": "stdout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "classname", 
          "failure": {
            "message": "failure message", 
            "type": "failure"
          }, 
          "jobid": null, 
          "name": "test-case-name-failure", 
          "system-out": {
            "text": "stdout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "classname", 
          "jobid": null, 
          "name": "test-case-name-skipped", 
          "system-out": {
            "text": "stdout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "class name", 
          "error": {
            "message": "error message", 
            "type": "error"
          }, 
          "jobid": null, 
          "name": "test-case-name-error-no- output", 
          "system-out": {
            "text": "std\nout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "class name", 
          "failure": {
            "message": "failure message", 
            "type": "failure"
          }, 
          "jobid": null, 
          "name": "test-case-name-failure-no- output", 
          "system-out": {
            "text": "std\nout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "class name", 
          "jobid": null, 
          "name": "test-case-name-skipped-no- output", 
          "system-out": {
            "text": "std\nout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "classname", 
          "error": {
            "type": "error"
          }, 
          "jobid": null, 
          "name": "test-case-name-error-no-message", 
          "system-out": {
            "text": "stdout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "classname", 
          "failure": {
            "type": "failure"
          }, 
          "jobid": null, 
          "name": "test-case-name-failure-no-message", 
          "system-out": {
            "text": "stdout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "classname", 
          "jobid": null, 
          "name": "test-case-name-skipped-no-message", 
          "system-out": {
            "text": "stdout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "classname", 
          "error": {
            "message": "error message", 
            "type": "error"
          }, 
          "jobid": null, 
          "name": "test-case-name-error-no-stderr", 
          "system-out": {
            "text": "stdout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "classname", 
          "failure": {
            "message": "failure message", 
            "type": "failure"
          }, 
          "jobid": null, 
          "name": "test-case-name-failure-no-stderr", 
          "system-out": {
            "text": "stdout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "classname", 
          "jobid": null, 
          "name": "test-case-name-skipped-no-stderr", 
          "system-out": {
            "text": "stdout"
          }, 
          "time": "1.200000"
        }, 
        {
          "classname": "classname", 
          "jobid": null, 
          "name": "optimal-success"
        }, 
        {
          "classname": "classname", 
          "error": {
            "message": "optimal error message", 
            "type": "error"
          }, 
          "jobid": null, 
          "name": "optimal-error"
        }, 
        {
          "classname": "classname", 
          "failure": {
            "message": "optimal failure message", 
            "type": "failure"
          }, 
          "jobid": null, 
          "name": "optimal-failure"
        }, 
        {
          "classname": "classname", 
          "jobid": null, 
          "name": "optimal-skipped"
        }
      ], 
      "tests": "17", 
      "time": "15.6", 
      "timestamp": "2017-03-03T20:30:50"
    }
  ], 
  "time": "15.6"
}
